todo-app/src/stores/TodosStore.js

```diff
@@ -0,0 +1,70 @@
+import todos from "../data/todos.json";
+import { observable, action, configure } from "mobx";
+
+configure({ enforceActions: true });
+
+
+const TodoItems = observable({
+
+    todos,
+
+    get nextId() {
+        if (!this.todos.length) {
+            return 1;
+        }
+        const todoIDs = this.todos.map(todo => todo.id);
+        return Math.max(...todoIDs) + 1
+    },
+
+    addTodo: action((id, description, deadline) => {
+        if (description && deadline) {
+            const newTodo = {
+                id,
+                description,
+                deadline,
+                done: false, 
+                editMode: false
```

Comment:
I see you've got for the "multiple" editing way instead of the single editing way. There's a small problem with embedding this kind of editing logic directly in our data - it will have to be added and removed each time we interface with an API. You'll have to do some extra refactoring next week.